SixStringsCoder/tracker_server
# Add dotenv for secure secrets and refactor signup route to encrypt password

## Description
- Add dotenv to secure secrets;
- Refactor the signup route to encrypt the password before saving at database and check if there is already an email registered;
- Add .gitignore.

## Type of change

## How to test
Use this curl to register a new user:
```
curl --location --request POST 'localhost:3000/auth/signup' \
--header 'Content-Type: application/json' \
--data-raw '{
    "name": "Leonardo Jesus",
    "email": "leonardojesus2020@outlook.com",
    "password": "123456"
}'
```

Response:
```
{
    "_id": "639133e869e51f1c1cb1046f",
    "name": "Leonardo Jesus",
    "email": "leonardojesus2020@outlook.com",
    "createdAt": "2022-12-08T00:46:32.897Z",
    "__v": 0
}
```

It should create a document similar to this on your database:
```
{"_id":{"$oid":"639133e869e51f1c1cb1046f"},"name":"Leonardo Jesus","email":"leonardojesus2020@outlook.com","password":"$2a$10$XcHwppObZoKoREnJw3itHusb50LRhATd766dkbR/1rlJ8xeZq3OW2","createdAt":{"$date":{"$numberLong":"1670460392897"}},"__v":{"$numberInt":"0"}}
```


## Artifacts
![image](https://user-images.githubusercontent.com/59581140/206330050-b956bf9b-2f48-4643-84d5-354439362ede.png)
